core/src/test/java/io/kestra/core/tasks/storages/LocalFilesTest.java

```diff
@@ -0,0 +1,96 @@
+package io.kestra.core.tasks.storages;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.runners.RunContextFactory;
+import io.kestra.core.storages.StorageInterface;
+import io.kestra.core.utils.IdUtils;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@MicronautTest
+class LocalFilesTest {
+    @Inject
+    RunContextFactory runContextFactory;
+
+    @Inject
+    StorageInterface storageInterface;
+
+    @Test
+    void run() throws Exception {
+        var runContext = runContextFactory.of(Map.of("toto", "tata"));
+        var existingFile = Files.createFile(Path.of(runContext.tempDir().toString(), "hello-output.txt"));
+        Files.write(existingFile, "Hello Output".getBytes());
+        var resource = ConcatTest.class.getClassLoader().getResource("application.yml");
+        var storageFile = storageInterface.put(
+            new URI("/file/storage/get.yml"),
+            new FileInputStream(Objects.requireNonNull(resource).getFile())
+        );
+
+        var task = LocalFiles.builder()
+            .id(IdUtils.create())
+            .type(LocalFiles.class.getName())
+            .inputs(List.of(
+                LocalFiles.LocalFile.builder().name("hello-input.txt").content("Hello Input").build(),
+                LocalFiles.LocalFile.builder().name("execution.txt").content("{{toto}}").build(),
+                LocalFiles.LocalFile.builder().name("application.yml").content(storageFile.toString()).build())
+            )
+            .outputs(List.of("hello-output.txt"))
+            .build();
+        var outputs = task.run(runContext);
+
+        assertThat(outputs, notNullValue());
+        assertThat(outputs.getOutputFiles(), notNullValue());
+        assertThat(outputs.getOutputFiles().size(), is(1));
+        assertThat(
+            new String(storageInterface.get(outputs.getOutputFiles().get(0)).readAllBytes()),
+            is("Hello Output")
+        );
+        assertThat(runContext.tempDir().toFile().list().length, is(3));
```

Comment:
I think it is missing a check on what was actually created in storage from the inputs as for eg. execution.txt contains a wrong pebble expression (toto instead of inputs.toto) ?

Reply:
Same for storage file as content which should result in a duplicate content inside the newly created file ?